packages/netlify-cms-core/src/actions/mediaLibrary.ts

```diff
@@ -200,7 +202,7 @@ export function persistMedia(file: File, opts: MediaOptions = {}) {
     const fileName = sanitizeSlug(file.name.toLowerCase(), state.config.get('slug'));
     const existingFile = files.find(existingFile => existingFile.name.toLowerCase() === fileName);
 
-    const editingDraft = selectEditingWorkflowDraft(state);
+    const editingDraft = selectEditingDraft(state.entryDraft);
```

Comment:
Media files will be persisted in a single commit with the entry regardless if you're using editorial workflow.